scheduler/planner.go

```diff
@@ -231,6 +235,10 @@ func (unit *Unit) RankValue() int64 {
 		// give generators a boost so people don't have to wait twice.
 		priority = priority * unit.distro.GetGenerateTaskFactor()
 	}
+	if stepbackTask {
+		// give stepback tasks a priority over mainline
```

Comment:
I'm actually thinking through this math a bit more. if my priority is initially 10:
For a mainline commit with factor 20, priority is now 200. Then when we get to MainlineCommitFactor, which let's say is 10, we're adding an overall priority of 2000 ish.
For patches with 10 initial priority, and a mainline commit with factor 20, we're adding an overall priority of 200 (so the same factor produces very different results).

All this to say, maybe we should handle stepback in the same line as mainline commits, maybe by adding to the mainline commit factor or etc.

Reply:
done